### ISSUE TITLE
PatternFly 6 upgrade preparation work

### ISSUE THREAD

**Issue by jelly** [cockpit-project/cockpit#20973]
Upgrading to PF6 will be a major task in all our Cockpit projects. This issue is here to collect everything which we can do upfront to make the bump to PF6 less painful:

* [x] Land https://github.com/cockpit-project/cockpit/pull/20962
* [x] The deprecated Select component has been removed in the [PF 6 branch](https://github.com/patternfly/patternfly-react/tree/v6/packages/react-core/src/deprecated/components) and is still used in [a lot of places](https://github.com/search?q=org%3Acockpit-project+%22deprecated%2Fcomponents%22&type=code)
     * [x] Remove comments about `noResultsFoundText`, this was removed  https://github.com/patternfly/patternfly-react/issues/6005
* [ ] Run [codemods if possible](https://github.com/patternfly/pf-codemods?tab=readme-ov-file#nav-remove-tertiary-variant-9948) for things which are removed or deprecated in PF6
     * [x] nav tertiary can become horizontal-subnav
     * [x] Switch labelOff https://github.com/patternfly/patternfly-react/pull/10646 https://github.com/cockpit-project/cockpit/pull/20978
     * [ ] All CSS is changed to pf-v5 => pf-v6 this is becoming unmaintainable churn :-)
* [ ] Go over our old PF4/5 issues and see if they are resolved.
     * [ ] Machines host vm table https://github.com/cockpit-project/cockpit-machines/blob/ef2bfbd336327e59a892c06bded8a7032f325a04/src/components/vms/hostvmslist.scss#L9 
     * [x] Hack which mentions it should be fixed in PF 5 https://github.com/cockpit-project/cockpit-machines/blob/ef2bfbd336327e59a892c06bded8a7032f325a04/src/components/vms/hostvmslist.scss#L26
     * [x] Cockpit => Clean up "# HACK: PF modal can have multiple id's in certain scenario's"
     * [ ] Cockpit https://github.com/patternfly/patternfly-react/issues/7592
     * [ ] Cockpit https://github.com/patternfly/patternfly/issues/4061
     * [x] Cockpit https://github.com/patternfly/patternfly/issues/9399
     * [ ] Cockpit https://github.com/patternfly/patternfly/issues/4524
     * [ ] Cockpit https://github.com/patternfly/patternfly/issues/5802
     * [ ] Cockpit https://github.com/patternfly/patternfly/issues/6643
     * [ ] Cockpit https://github.com/patternfly/patternfly-react/issues/5379
     * [ ] Cockpit https://github.com/patternfly/patternfly-react/issues/5606
     * [ ] Cockpit https://github.com/patternfly/patternfly/issues/4280
     * [ ] Cockpit https://github.com/patternfly/patternfly/pull/2757
     * [ ] Cockpit https://github.com/patternfly/patternfly/pull/3714
     * [ ] Cockpit https://github.com/patternfly/patternfly/pull/2921
 

More PF issues can be found with:
```
git grep "github.com/patternfly/patternfly"
```

**Comment by jelly** [comment]
Most of the codemods change code to renamed properties or changed implementations which we can't do upfront.

Instead of changing all our tests to handle `pf-v5-c` -> `pf-v6-c` we could write our tests like:

```
test/verify/check-users:        b.wait_visible("#accounts div.pf-v5-c-empty-state")
```

```
test/verify/check-users:        b.wait_visible(f"#accounts div.{self.pf_version}-c-empty-state")
```

**Comment by jelly** [comment]
nav tertiary looks :100: different from the horizontal subnav. The screenshot is stretched because of the Flex we have there. 
 
![image](https://github.com/user-attachments/assets/655b7ec5-8ba7-44bf-a15f-1809aaa16d35)

**Comment by garrett** [comment]
> nav tertiary looks 💯 different from the horizontal subnav. The screenshot is stretched because of the Flex we have there.

Yep, it's not a replacement, regardless of what they've said. The visual replacement would be tabs, but they should activate a location instead of switch a tab notebook.

In other words, I think this is what we'd want instead (tabs as navigation): https://www.patternfly.org/components/tabs/#tabs-linked-to-nav-elements

**Comment by martinpitt** [comment]
https://github.com/cockpit-project/cockpit-podman/pull/1951 was the last puzzle piece of deprecated PF components.

**Comment by jelly** [comment]
@martinpitt yup, I believe we have done the most that can be done here so we can probably close it.
